frontend/src/components/user/VendingLink.tsx

```diff
@@ -0,0 +1,169 @@
+import { useTranslation } from "next-i18next"
+import { FunctionComponent, useEffect } from "react"
+import { toast } from "react-toastify"
+import {
+  VENDING_DASHBOARD_URL,
+  VENDING_ONBOARDING_URL,
+  VENDING_STATUS_URL,
+} from "../../env"
+import { useAsync } from "../../hooks/useAsync"
+import { VendingRedirect, VendingStatus } from "../../types/Vending"
+import Button from "../Button"
+import Spinner from "../Spinner"
+import styles from "./VendingLink.module.scss"
+
+// API responds with this status code if vending has not begun
+const STATUS_NEW = 201
+
+/**
+ * Retrieve the vending status of the logged in user.
+ * @returns The vending status object, or nothing if onbording has not started.
+ */
+async function getStatus() {
+  let res: Response
+  try {
+    res = await fetch(VENDING_STATUS_URL, { credentials: "include" })
+  } catch {
+    throw "failed-to-load-refresh"
+  }
+
+  if (res.ok) {
+    if (res.status === STATUS_NEW) {
+      return
```

Comment:
Not a fan of functions that return vastly different results. At least annotate it correctly with `Promise<VendingStatus | undefined>` as typescript doesn't seem to pick up on that correctly (with the current config)